package/src/lib/octokit.ts

```diff
@@ -1,43 +1,111 @@
+      } else if (type === "warn") {
+        console.log(`[WARN] [astro-gists : octokit] ${message}`);
+      } else if (type === "error") {
+        console.log(`[ERROR] [astro-gists : octokit] ${message}`);
+      } 
+    }
 
-export const TOKEN_MISSING_ERROR = "GITHUB_PERSONAL_TOKEN not found. Please add it to your .env file. Without it, you will be limited to 60 requests per hour.";
+    if (!VerboseCheck) {
+      if (type === "info") {
+        console.log(`[astro-gists : octokit]" ${message}`);
+      } else if (type === "warn") {
+        console.log(`[WARN] [astro-gists : octokit] ${message}`);
+      } else if (type === "error") {
+        console.log(`[ERROR] [astro-gists : octokit] ${message}`);
+      }
+    }
+
+  };
+
+// Load environment variables
+const { GITHUB_PERSONAL_TOKEN } = loadEnv("all", process.cwd(), "GITHUB_");
 
 // Create an Octokit instance
 const octokit = new Octokit({ auth: GITHUB_PERSONAL_TOKEN });
 
-// Retry requests if rate limited
-export async function requestRetry(route: Route, parameters: RequestParameters) {
-  try {
-    const response: OctokitResponse<unknown, number> = await octokit.request(route, parameters);
-    return response;
-  } catch (error) {
-    /** @ts-ignore-error */
-    if (error.response && error.status === 403 && error.response.headers['x-ratelimit-remaining'] === '0') {
-      /** @ts-ignore-error */
-      const resetTimeEpochSeconds = error.response.headers['x-ratelimit-reset'];
-      const currentTimeEpochSeconds = Math.floor(new Date().getTime() / 1000);
-      const secondsToWait = resetTimeEpochSeconds - currentTimeEpochSeconds;
-      console.log(`Rate limit reached. Waiting ${secondsToWait} seconds before retrying.`);
-      return new Promise((resolve) => {
-        setTimeout(async () => {
-          const retryResponse = await requestRetry(route, parameters);
-          resolve(retryResponse);
-        }, secondsToWait * 1000);
-      });
-    }
-      // Return a rejected Promise
-      return Promise.reject(error);
+// Retry failed requests
+const retry: typeof pRretry = (fn, opts) =>
+  pRretry(fn, {
+    onFailedAttempt: (e) =>
+      gistLogger("warn",
+        `Attempt ${e.attemptNumber} failed. There are ${e.retriesLeft} retries left.\n ${e.message}`,
+        false),
+      retries: 3,
+    ...opts,
+  });
+
+// Handle the response from the Octokit API
+// biome-ignore lint/suspicious/noExplicitAny: any is used to handle the response from the Octokit API
+function handleResponse(response: OctokitResponse<any>) {
+  switch (response.status) {
+    case 200:
+      return response.data;
+    case 404:
+      return "E404";
+    case 403:
+      return "E403";
+    case 500:
+      return "E500";
+    default:
+      return "E000";
+  }
+}
+// Gist Grabber
+const gistGrabber = async (gistId: string) => { 
+  const response = await retry(() => octokit.request('GET /gists/{gist_id}', { gist_id: gistId }));
+  if (handleResponse(response) === "E404") {
```

Comment:
Could store the status code in a const

`const statusCode = getStatusCode(response)`